### PR TITLE
navigate via keyboard

### PR DESCRIPTION
**Pull request by kiok46**
Now user can navigate in the new project dialog box and open the projects by keyboard.
Press **'up'**       -> to go up
Press **'down'**  -> to go down
Press **'enter'**  -> to open the project

### CODE REVIEW

File: designer/new_dialog.py [hunk 1]
```diff
@@ -77,6 +77,8 @@ def __init__(self, **kwargs):
         super(NewProjectDialog, self).__init__(**kwargs)
         item_strings = list(NEW_PROJECTS.keys())
         item_strings.sort()
+        self._keyboard = Window.request_keyboard(self._keyboard_closed, self)
```
Comment (aron-bordin) [inline designer/new_dialog.py]: Use the `Window.bind(on_key_down..` here. It's already being used by shortcut manager so we can have more control.

File: designer/new_dialog.py [hunk 2]
```diff
@@ -89,6 +91,24 @@ def __init__(self, **kwargs):
         self.list_parent.add_widget(self.listview, 1)
         self.on_adapter_selection_change(self.adapter)
 
+    def _keyboard_closed(self):
+        self._keyboard.unbind(on_key_down=self._on_keyboard_down)
+        self._keyboard = None
+
+    def _on_keyboard_down(self, keyboard, keycode, text, modifiers):
```
Comment (aron-bordin) [inline designer/new_dialog.py]: return True to stop the event propagation when the keycode matches

File: designer/new_dialog.py [hunk 3]
```diff
@@ -89,6 +91,24 @@ def __init__(self, **kwargs):
...
+        '''To detect which key is pressed
+        '''
```
Comment (aron-bordin) [inline designer/new_dialog.py]: check if `modifiers` is empty here, maybe I'm trying to run some shortcut ([ctrl] + up, [ctrl, shift] + down, etc) and these shortcuts cannot interact here.

**Comment by aron-bordin** [comment]
Hi @kiok46 ,

Please, check the comments above. Also, the NewProjectDialog is created each time it's opened, so instead of binding on __init__ you can use the `NewProjectDialog.on_parent` to bind/unbind the key_down event

**Comment by kiok46** [comment]
@aron-bordin  I made some changes, please check if they are good.
And also modifier is empty here.

**Comment by aron-bordin** [comment]
Hi @kiok46 
The modifiers identify if you are holding some modifier key (ctrl, shift, etc), so in this case it's necessary to ensure that `modifiers` variable is empty. just add a `if modifiers: \n return False` in the beginning of the key_down callback.

Also, this class still listeing the keyboard after closing the dialog. So it's better to unbind it while not using. You can remove the bind from `__init__`, add a new method `on_parent` and bind/unbind checking if the current dialog is attached do a parent widget or not.

**Comment by kiok46** [comment]
@aron-bordin  Will do (y)

**Comment by aron-bordin** [comment]
@kiok46 Great work, thx for contributing :+1:

**Comment by kiok46** [comment]
@aron-bordin  You're welcome :smile:

**Pull request merged**